http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -68,6 +78,22 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
 
       _ <- checkIfUserAllowed(req.appType, userInfo.userEmail)
 
+      projectLabels <- googleResourceService.getLabels(googleProject)
+      _ <-
+        if (req.appType == Custom && customAppSecurityConfig.enableCustomAppCheck) {
```

Comment:
we usually prefer to use the `match` keyword to simplify complex branches.

I believe, simplified, it would look something like this

```
appAllowList =
            if (projectLabels.getOrElse("security-group", "").canEqual("high"))
              customAppSecurityConfig.customApplicationAllowList.highSecurity
            else customAppSecurityConfig.customApplicationAllowList.default
req.appType match {
   case Custom if customAppSecurityConfig.enableCustomAppCheck => ...
   case Custom if !appAllowList.contains(req.descriptorPath.getOrElse("None").toString) => ...
   case _ => F.unit
}
```